File: KnowledgeCenterServer/KnowledgeCenter.DataConnector/Entities/Match/CustomerOffer.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;

namespace KnowledgeCenter.DataConnector.Entities.Match
{
    [Table("Match.CustomerOffer")]
    public class CustomerOffer
    {
        public int Id { get; set; }
        public string JobTitle { get; set; }
        public string Requester { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime ModificationDate { get; set; }
        public DateTime MissionStartDate { get; set; }
        public DateTime? MissionEndDate { get; set; }
        public bool MobilityRequired { get; set; }
        public bool OnSite { get; set; }
        public bool WorkFromHome { get; set; }
        public int CustomerOfferStatusId { get; set; }
        public CustomerOfferStatus CustomerOfferStatus { get; set; }
        public string Description {get; set;}
        public int CustomerAccountManagerId { get; set; }
        public User CustomerAccountManager { get; set; }

        public int CustomerSiteId { get; set; }
        public CustomerSite CustomerSite { get; set; }
        public ICollection<CustomerOfferSkill> CustomerOfferSkills { get; set; }
    }
}

File: KnowledgeCenterClient/src/app/match/models/CustomerOffer.ts
import { CustomerOfferStatus } from './CustomerOfferStatus';
import { User } from '../../shared/models/User';
import { CustomerSite } from './CustomerSite';
import { CustomerOfferSkill } from './CustomerOfferSkill';

export interface CustomerOffer {
  id: number;
  jobTitle: string;
  requester: string;
  creationDate: Date;
  missionStartDate: Date;
  missionEndDate: Date;
  mobilityRequired: boolean;
  onSite: boolean;
  workFromHome: boolean;
  customerOfferStatusId: number;
  customerOfferStatus: CustomerOfferStatus;
  description: string;
  customerAccountManagerId: number;
  customerAccountManager: User;
  customerSiteId: number;
  customerSite: CustomerSite;
  customerOfferSkills: CustomerOfferSkill[];
}
